autodiff/variable.py

```diff
@@ -1,24 +1,63 @@
 import numpy as np
 from autodiff.utils import get_right_shape
-#from variable import Variable
+
 
 class Variable:
     """ 
-    #TODO: Write some documentation about that.
-    Variable class carry the information flow within the computational graph. 
-    Data attribute, Gradient attribute.
-    Data represents the evaluation point and gradient is the gradient held 
-    by the variable. By default = 1. 
+    Represents a variable, and carries the information flow
+    within the computational graph. Under the current 
```

Comment:
A Variable is an object, not a scalar. It has two attributes val, grad that are numpy arrays (made of np.float64 type).
Sor far, it is supopsed to be instantiated with  a numeric type (e.g float, int) but it handles np.array directly, or even a 1-dim tuple or 1-d list. 
e.g : x = Variable(2.) 
Can handle x = Variable([2]) or x = Variable((2)) or x = Variable(np.array([2])).